2022Swerve/src/main/java/frc/robot/subsystems/AutonSubsytem.java

```diff
@@ -0,0 +1,198 @@
+package frc.robot.subsystems;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+
+import edu.wpi.first.math.controller.PIDController;
+import edu.wpi.first.math.controller.ProfiledPIDController;
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.math.trajectory.Trajectory;
+import edu.wpi.first.math.trajectory.TrajectoryConfig;
+import edu.wpi.first.math.trajectory.TrajectoryGenerator;
+import edu.wpi.first.math.trajectory.TrajectoryUtil;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.util.datalog.DataLog;
+import edu.wpi.first.util.datalog.StringLogEntry;
+import edu.wpi.first.wpilibj.DataLogManager;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.Filesystem;
+import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
+import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.smartdashboard.SendableChooser;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.InstantCommand;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.SwerveControllerCommand;
+import edu.wpi.first.wpilibj2.command.WaitCommand;
+import frc.robot.Constants.AutonConstants;
+import frc.robot.Constants.DriveConstants;
+
+
+public class AutonSubsytem extends SubsystemBase{
+    private enum AutonModes{
+        FORWARD, // Go forward 2 meters
+        BACKWARD, // Wait 3 seconds, go backward 2 meters
+        FORWARD180; // Go forward 2 meters and rotate 180 degrees
+    }
+
+    private ShuffleboardTab configTab = Shuffleboard.getTab("Config");
+    private NetworkTableEntry delayEntry = configTab.add("Auton Delay", 0.0).getEntry();
+    private SendableChooser<AutonModes> modeChooser = new SendableChooser<>();
+
+    private DataLog datalog = DataLogManager.getLog();
+    private StringLogEntry trajectoryLog = new StringLogEntry(datalog, "/auton/trajectory"); //Logs x translation state output
+    private StringLogEntry commandLog = new StringLogEntry(datalog, "/auton/command"); //Logs x translation state output
+
+    private SwerveSubsystem swerveSubsystem;
+
+    private PIDController xController;
+    private PIDController yController;
+    private ProfiledPIDController rotationController;
+
+    private TrajectoryConfig trajectoryConfig;
+
+    private AutonModes autonMode;
+
+    public AutonSubsytem(SwerveSubsystem swerveSubsystem){
+        this.swerveSubsystem = swerveSubsystem;
+
+        //Add auton modes to chooser
+        for(AutonModes mode : AutonModes.values()){
+            modeChooser.addOption(mode.toString(), mode);
```

Comment:
Add [setDefaultOption](https://first.wpi.edu/wpilib/allwpilib/docs/release/java/edu/wpi/first/wpilibj/smartdashboard/SendableChooser.html#setDefaultOption(java.lang.String,V)) for modeChooser. Otherwise Auton mode may crash if we forget to select something